pulsar-client-tools/src/main/java/com/yahoo/pulsar/admin/cli/PulsarAdminTool.java

```diff
@@ -108,8 +121,28 @@ boolean run(String[] args) {
             String cmd = jcommander.getParsedCommand();
             JCommander obj = jcommander.getCommands().get(cmd);
             CmdBase cmdObj = (CmdBase) obj.getObjects().get(0);
+            if (StringUtils.isNotBlank(serviceUrl) || (StringUtils.isNotBlank(authPluginClassName) && StringUtils
+                    .isNotBlank(authParams))) {
+                if (StringUtils.isNotBlank(serviceUrl)) {
+                    try {
+                        url = new URL(serviceUrl);
+                    } catch (MalformedURLException e) {
+                        System.err.println("Invalid serviceUrl: '" + serviceUrl + "'");
+                        System.exit(1);
+                    }
+                }
+                try {
+                    if (StringUtils.isNotBlank(authPluginClassName) && StringUtils.isNotBlank(authParams)) {
+                        config.setAuthentication(authPluginClassName, authParams);
+                    }
+                    cmdObj = cmdObj.getClass().getConstructor(PulsarAdmin.class).newInstance(new PulsarAdmin(url, config));
```

Comment:
I think it's not better to parse args in main method in view of OOP.
However, It's not better to recreate `PulsarAdmin` and `CmdBase` either.

To resolve it, I changed the code as follows.

1. In constructor, objects of `PulsarAdmin` and classes implementing `CmdBase` are not created.  
   Instead of it, commands are put to `commandMap` which is an object of Map<String, Class>
2.  The commands are added to `jcommander` by `setupCommands` method when showing help or running commands.